src/Middleware/Buffer.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace React\Http\Middleware;
+
+use Psr\Http\Message\ServerRequestInterface;
+use React\Http\Response;
+use React\Promise\Stream;
+use React\Stream\ReadableStreamInterface;
+use RingCentral\Psr7\BufferStream;
+
+final class Buffer
+{
+    private $sizeLimit;
+
+    public function __construct($sizeLimit = null)
+    {
+        if ($sizeLimit === null) {
+            $sizeLimit = $this->iniMaxPostSize();
+        }
+
+        $this->sizeLimit = $sizeLimit;
+    }
+
+    public function __invoke(ServerRequestInterface $request, $stack)
+    {
+        $size = $request->getBody()->getSize();
+
+        if ($size === null) {
+            return new Response(411, array('Content-Type' => 'text/plain'), 'No Content-Length header given');
+        }
```

Comment:
Makes perfect sense to me, but we should probably add some documentation and tests for this :+1: 

For the reference: I've discussed this with @WyriHaximus and I think it's save to say we all agree that it may make sense to also support incoming requests with `Transfer-Encoding: chunked` in a future version (follow-up PR) :+1: This feature is rarely used in the real world and not something that will be needed for most people (unlike outgoing responses with `Transfer-Encoding: chunked` which are supported).